### PR TITLE
feat: Configure target ECS cluster ARN

### LINKED ISSUE #13: Support cluster selection
It would be nice if the clusters to check were configurable. Exposing _all_ clusters on ECS through the same mechanisms might not be desireable at all times.

I'm working on this, but my Python skills are poor 😄

### PR DESCRIPTION
**Pull request by oliversalzburg**
Allows configuring a specific ECS cluster to monitor, using the `--cluster=ARN` CLI argument.

Fixes #13

If someone could point out how to adjust the log line during startup to include this setting, I'd be happy to include it.

### CODE REVIEW

File: discoverecs.py
Comment (Limess) [inline discoverecs.py]: This would be more useful as a list so multiple clusters can be used

File: discoverecs.py
Comment (Limess) [inline discoverecs.py]: See above, it'd be better if this is a list. 

It'd also be better to be explicit and call it `--cluster-arns`, right now it's unclear whether we mean cluster name in the same account, or the full arn.

I've just made a change to simplify the log at the start and add an option which uses a list if you want to try to follow that example.

File: discoverecs.py
Comment (Limess) [inline discoverecs.py]: Keyword arguments would be preferable here

```suggestion
        self.discoverer = TaskInfoDiscoverer(cluster_arns=cluster_arns)
```

**Review by Limess — CHANGES_REQUESTED** [review]
Thanks for the contribution!

We don't spend a lot of time maintaining this or checking for notifications, but this looks like a sensible improvement so I'll try to help you get it in.

It'd be useful if you were to also update the README with the new option. I'm aware that's a little bit messy right now so don't worry about placement too much.

File: discoverecs.py
Comment (oliversalzburg) [reply discoverecs.py]: I agree and originally planned to do this, but my Python skills are very poor and I have a hard time testing the script. I will try my best to make the appropriate changes. Thanks for the help.

File: discoverecs.py
Comment (oliversalzburg) [reply discoverecs.py]: That looks very helpful. I will rebase my change on the latest HEAD to be able to use all of this. I hope it doesn't mess up the review

File: discoverecs.py
Comment (oliversalzburg) [reply discoverecs.py]: Awesome. Thanks

**Comment by oliversalzburg** [comment]
I've made the changes to the best of my knowledge. I'll test these in my own setup and fix any issues that come up. Otherwise, I'd also appreciate a second review anytime :)

File: discoverecs.py
Comment (Limess) [inline discoverecs.py]:
```suggestion
                if len(self.cluster_arns) > 0 and cluster_arn not in self.cluster_arns:
```

This should continue to work as without the argument if no cluster arns are specified

**Review by Limess — CHANGES_REQUESTED** [review]
Another small change to ensure the current behaviour is the same but looks good

File: discoverecs.py
Comment (oliversalzburg) [reply discoverecs.py]: I thought about this while writing that line. I figured, since it's an empty list by default, `cluster_arn` won't be in it. That's why I felt a length-check was redundant.

I'm happy to go with your suggestion regardless, if you prefer it.

Comment (Limess) [reply discoverecs.py]: I just tested the current implementation locally and it doesn't find any tasks when no cluster arns are specified, pretty sure the length check is required else we skip every cluster

Comment (Limess) [reply discoverecs.py]: It's probably preferable to add a `list_clusters` method, which returns `self.cluster_arns` if it exists, otherwise iterates as the current behavior, and then call that in `get_infos`. I don't think there's anything to be gained by paginating through clusters if you already know what you're looking for.

i.e.

```python
    def list_clusters(self):
        if self.cluster_arns:
			return self.cluster_arns
        cluster_arns = []
        clusters_pages = self.ecs_client.get_paginator("list_clusters").paginate()
        for clusters in clusters_pages:
            for cluster_arn in clusters["clusterArns"]:
            	cluster_arns += cluster_arn
		return cluster_arns
```

Comment (oliversalzburg) [reply discoverecs.py]: Huh, I see. Thanks for the suggestion. I really like that idea. Sadly, I'm having a hard time getting it to work. I'm currently working with this function:

```python
    def list_clusters(self):
        if len(self.cluster_arns) > 0:
            return self.cluster_arns
        cluster_arns = []
        clusters_pages = self.ecs_client.get_paginator("list_clusters").paginate()
        for clusters in clusters_pages:
            for cluster_arn in clusters["clusterArns"]:
                cluster_arns += cluster_arn

        print(cluster_arns)
        return cluster_arns
```

In the case where I don't specify ARNs explicitly, I expect `print(cluster_arns)` to print a list of the ARNs of my ECS clusters. However, it's a list of the individual characters of those ARNs. My attempts to resolve this were not successful 😞

This leads to failures down the line. I figured the problem needs to be solved in this function already.

Comment (oliversalzburg) [reply discoverecs.py]: And just after writing this, I find a resource that shows `cluster_arns += [cluster_arn]` as a way to append a string to a list of strings. I assume this is the way to go?

Comment (Limess) [reply discoverecs.py]: Ah sorry yes that's a mistake in my suggestion (I don't write python often either). The cluster_arn in my suggestion is being treated as a list of characters 🤦🏻‍♂️ 

You can either do as you suggested: `cluster_arns += [cluster_arn]`, or `append(cluster_arns, cluster_arn)`, either are fine and should work.

Comment (oliversalzburg) [reply discoverecs.py]: Okay, awesome. I did that in my latest commit 😊

File: discoverecs.py
Comment (Limess) [inline discoverecs.py]: Remove this print and then I'll try to merge this when I have the time 👍🏻

Comment (oliversalzburg) [reply discoverecs.py]: Alright. Done. Thanks for the review :)

**Comment by Limess** [comment]
Merged, thanks again!

**Pull request merged**
